tree.go

```diff
@@ -428,7 +481,139 @@ func (n *InternalNode) Serialize() ([]byte, error) {
 	return rlp.EncodeToBytes([]interface{}{bitlist, children})
 }
 
+func (n *accountLeaf) Insert(k []byte, value []byte) error {
+	// The parent insert is expected to ensure that this
+	// situation doesn't occur. This check will catch an
+	// invalid situation while the library stabilizes.
+	if !bytes.Equal(k[:31], n.key[:31]) {
+		return errors.New("inserting invalid key into key")
+	}
+
+	switch k[31] {
+	case 1:
+		n.Balance.SetBytes(value)
+		n.commitment = nil // invalidate commitment
+	case 2:
+		n.Nonce = binary.BigEndian.Uint64(value)
+		n.commitment = nil // invalidate commitment
+	default:
+		return errors.New("writing to read-only location")
+	}
+	return nil
+}
+
+func (n *accountLeaf) CreateAccount(k []byte, version, nonce, codeSize uint64, balance *big.Int, codeHash common.Hash) error {
+	// The parent insert is expected to ensure that this
+	// situation doesn't occur. This check will catch an
+	// invalid situation while the library stabilizes.
+	if !bytes.Equal(k[:31], n.key[:31]) {
+		return errors.New("inserting invalid key into key")
+	}
+
+	n.key = k
+	n.Version = version
+	n.Balance = balance
+	n.Nonce = nonce
```

Comment:
note to self: does it make sense to set the nonce from the start? This function is trying to realize both "create" when an account is created, and "insert" when an account is loaded into the tree. For a prototype, it's fine. A note should be taken, though.